ui/jobs/jobs.js

```diff
@@ -2108,6 +2169,132 @@ class Bars {
+      disembowelBox.valuescale = this.gcdSkill();
+      disembowelBox.threshold = this.gcdSkill() * 5;
+      highJumpBox.valuescale = this.gcdSkill();
+      highJumpBox.threshold = this.gcdSkill() + 1;
+    };
+    // ComboTimer Bar
+    // DRG's skill effect buff appear on self at once,
+    // but start counting down when damage is deal.
+    // This cause some timer not accurate.
+    const comboTimer = this.addTimerBar({
+      id: 'drg-timers-combo',
+      fgColor: 'drg-color-combo',
+    });
+    let comboType = '';
+    const buffComboFunc = () => {
+      comboTimer.duration = 0;
+      comboTimer.duration = 10;
+      comboType = 'buff';
+    };
+    const comboLostFunc = () => {
+      comboTimer.duration = 0;
+    };
+
+    this.comboFuncs.push((skill) => {
+      // if this.combo.considerNext is equal as this.combo.startList, the combo is broken.
+      if (this.combo.considerNext == this.combo.startList)
+        comboTimer.duration = 0;
+
+      // if skill exist, this skill is in combo.
+      if (skill) {
+        comboTimer.duration = 0;
+        comboTimer.duration = 15;
+        comboType = 'normol';
+
+        if (skill == kAbility.Disembowel) {
+          disembowelBox.duration = 0;
+          disembowelBox.duration = 30 + 1;
+        }
+        // When Sonic Thrust hit more than one mob, combo will be errorly break.
+        // ComboTracker is hard to fix, so deal it by a tricky way.
+        if (skill == kAbility.SonicThrust) {
+          setTimeout(() => {
+            comboTimer.duration = 0;
+            comboTimer.duration = 14.9;
+          }, 100);
+        }
+      }
+    });
+
+    [
+      kAbility.PiercingTalon,
+      kAbility.FullThrust,
+      kAbility.ChaosThrust,
+      kAbility.CoerthanTorment,
+      kAbility.FangAndClaw,
+      kAbility.WheelingThrust,
+    ].forEach((ability) => {
+      this.abilityFuncMap[ability] = comboLostFunc;
+    });
+
+    [
+      EffectId.SharperFangAndClaw,
+      EffectId.EnhancedWheelingThrust,
+      EffectId.RaidenThrustReady,
+    ].forEach((effectId) => {
+      this.gainEffectFuncMap[effectId] = buffComboFunc;
+    });
+
+    // Gauge
+    const blood = this.addResourceBox({
+      classList: ['drg-color-blood'],
+    });
+    const eyes = this.addResourceBox({
+      classList: ['drg-color-eyes'],
+    });
+    this.jobFuncs.push((jobDetail) => {
+      blood.parentNode.classList.remove('blood', 'life');
+      if (jobDetail.bloodMilliseconds > 0) {
+        blood.parentNode.classList.add('blood');
+        blood.innerText = Math.ceil(jobDetail.bloodMilliseconds / 1000);
+        if (jobDetail.bloodMilliseconds < 5000)
+          blood.parentNode.classList.remove('blood');
+      } else if (jobDetail.lifeMilliseconds > 0) {
+        blood.parentNode.classList.add('life');
+        blood.innerText = Math.ceil(jobDetail.lifeMilliseconds / 1000);
+      } else {
+        blood.innerText = '';
+        if (comboType == 'buff')
```

Comment:
Infact there's another issue. 
If the last combo hit the right direction, RaidenThrustReady will be triggered and combo continue.
(In other words if DRG always hit right direction, combo circle endlessly.)
But if you hit wrong derection, combo should break, which makes issue more complex.

Reply:
Yeah, I don't know what to do about RaidenThrust here.  I think it might be easier to just "break" the combo at the end like it does now.

You could argue that it's not technically a combo, it's a buff? RaidenThrust runs out in 10s, but your combo runs out in 15s.